src/main/grammars/RustParser.bnf

```diff
@@ -703,8 +703,9 @@ private TypeReferenceInner ::= ArrayType
                                // External rule to find the difference between `(T)` and `(T,)`
                                | <<tupleOrParenType TypeReferenceInner TupleType_upper>>
                                | ExplicitTraitType
+                               | TrivialBaseType
                                // External rule to find the difference between `T` and `T+T`
-                               | <<baseOrTraitType BaseTypeInner ImplicitTraitType TraitType_upper>>
+                               | <<baseOrTraitType PathGenericArgsWithoutColons ImplicitTraitType TraitType_upper>>
```

Comment:
It would be great to parse `() + Send` too and add error annotation via some inspection
But I think it can be done in other PR - looks like that it will require a lot of changes

Reply:
Hm, we steel parse `FnOnce() -> i32 + Send` as `FnOnce` with `i32 + Send` ret type